org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/ProgressReporterManagerTest.java

```diff
@@ -106,6 +107,23 @@ protected IStatus run(IProgressMonitor monitor) {
 		assertTrue(report.isComplete());
 	}
 
+	@Test
```

Comment:
I added a unit test but I cannot run the test. It fails with following error:

```
Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.jdt.launching.IVMInstallType.findVMInstall(String)" because "vmInstallType" is null
	at org.eclipse.jdt.ls.core.internal.TestVMType.setTestJREAsDefault(TestVMType.java:49)
	at org.eclipse.jdt.ls.core.internal.JavaLanguageServerTestPlugin.start(JavaLanguageServerTestPlugin.java:36)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:818)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:1)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:810)
	... 24 more
```

Could you give me some hint what I'm doing wrong? I tried `./mvnw install -Dtest=ProgressReporterManagerTest -DfailIfNoTests=false`

Reply:
The test works for me :)

`-Dtest` would work for maven-surefire. However, this isn't a JUnit test, it's a JUnit (Eclipse) Plugin Test. So it runs the tests with the tycho-surefire plugin. As a result, the property I use is `testClass` from https://www.eclipse.org/tycho/sitedocs/tycho-surefire-plugin/test-mojo.html#testClass .

`mvn clean verify -DtestClass=org.eclipse.jdt.ls.core.internal.handlers.ProgressReporterManagerTest -DfailIfNoTests=false`

You could also just run the full test suite with `mvn clean verify` though that takes a bit longer.